scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -82,7 +82,7 @@ class FossId(
 
     companion object {
         @JvmStatic
-        private val PROJECT_NAME_REGEX = Regex("""^.*\/([\w\-]+)\.git$""")
+        private val PROJECT_NAME_REGEX = Regex("""^.*\/([\w\-.]+)(?:\.git)*$""")
```

Comment:
Why does this require a literal `.` to be accepted as part of the project name? If it's really required, and you adapt the example URL accordingly?

Also, would the suffix just become `(\.git)?`?

Reply:
Commit message title:
- s/git/Git/
- s/url/URL/

Also, it's it the "*project name* pattern" you're adjusting, not the "repo url pattern"?

Reply:
Commit message body:
- s/URLS/URLs/
- s/like/like:/

Reply:
> Why does this require a literal `.` to be accepted as part of the project name? If it's really required, and you adapt the example URL accordingly?

this seems like a mistake. Removed.
> 
> Also, would the suffix just become `(\.git)?`?

this part of your comment I don't understand ?!

Reply:
Ha yes I understand. `?` is more precise than `*`, clear. But I want to keep the non capturing group.

Reply:
> But I want to keep the non capturing group.

Why? Personally, I find non-capturing groups a bit harder to understand.

Reply:
This a good practice not ? You capture only if you intend something with it ?
For instance:
https://www.reddit.com/r/regex/comments/kuapwm/should_i_use_noncapturing_groups_if_i_dont_care/

Reply:
Regarding your comment

> Also, it's it the "project name pattern" you're adjusting, not the "repo url pattern"?

The pattern is meant to be matched against the Git repo URL.
The fact that it is also used to extract the project name is another aspect of it, but IMO no relevant here.

Reply:
> The pattern is meant to be matched against the Git repo URL.

I know, but I what I meant was that's it's probably easier to understand to refer to the *variable name*, rather than to the string it matches against. But this was just a suggestion / thought. I'm fine with keep this as-is.

Reply:
Why not. Done.